crates/pypi-types/src/simple_json.rs

```diff
@@ -150,31 +130,34 @@ pub struct Hashes {
 }
 
 impl Hashes {
-    /// Format as `<algorithm>:<hash>`.
-    pub fn to_string(&self) -> Option<String> {
-        self.sha512
-            .as_ref()
-            .map(|sha512| format!("sha512:{sha512}"))
-            .or_else(|| {
-                self.sha384
-                    .as_ref()
-                    .map(|sha384| format!("sha384:{sha384}"))
-            })
-            .or_else(|| {
-                self.sha256
-                    .as_ref()
-                    .map(|sha256| format!("sha256:{sha256}"))
-            })
-            .or_else(|| self.md5.as_ref().map(|md5| format!("md5:{md5}")))
-    }
-
-    /// Return the hash digest.
-    pub fn as_str(&self) -> Option<&str> {
-        self.sha512
-            .as_deref()
-            .or(self.sha384.as_deref())
-            .or(self.sha256.as_deref())
-            .or(self.md5.as_deref())
+    /// Convert a set of [`Hashes`] into a list of [`HashDigest`]s.
+    pub fn into_digests(self) -> Vec<HashDigest> {
+        let mut digests = Vec::new();
+        if let Some(md5) = self.md5 {
+            digests.push(HashDigest {
+                algorithm: HashAlgorithm::Md5,
+                digest: md5,
+            });
+        }
```

Comment:
Does inserting this first cause a change in behavior? It looks like `as_str()` previously preferred `sha512` for example, but I think I saw some calls to `hashes.first()` above which would prefer `md5` I think?

Reply:
Hmm, I guess it _could_ yes. I'll change it to put the highest-security hashes first.